testing/machine/secret.go

```diff
@@ -63,6 +64,13 @@ func (m *Machine) WaitForSecret(namespace string, name string) error {
 	})
 }
 
+// CustomWaitForSecret blocks until the secret is available. It fails after the custom timeout.
```

Comment:
nit: I wasn't sure what was 'custom' about this at first, but  `WaitForSecretWithTimeout` seems to long. If it's only used in one place, maybe we don't need a helper. `Eventually` might also help.